From b84296a431530a055d444b20366004cfb37271c4 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 10 Jul 2017 12:38:03 +0800
Subject: [PATCH 2101/5242] MLK-15932-5 gpu: imx: dpu: common: Introduce
 dpu_vproc_has_h/vscale_cap()

commit  5227c0f1090a311bdd5cd6a8e1d380af3d6d7099 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch introduces two helpers - dpu_vproc_has_h/vscale_cap().

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |   12 ++++++++++++
 include/video/dpu.h              |    3 +++
 2 files changed, 15 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index 68af5ae..171d00e 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -520,6 +520,18 @@ static inline void dpu_cm_write(struct dpu_soc *dpu, u32 value,
 };
 MODULE_DEVICE_TABLE(of, dpu_dt_ids);
 
+bool dpu_vproc_has_hscale_cap(u32 cap_mask)
+{
+	return !!(cap_mask & DPU_VPROC_CAP_HSCALE);
+}
+EXPORT_SYMBOL_GPL(dpu_vproc_has_hscale_cap);
+
+bool dpu_vproc_has_vscale_cap(u32 cap_mask)
+{
+	return !!(cap_mask & DPU_VPROC_CAP_VSCALE);
+}
+EXPORT_SYMBOL_GPL(dpu_vproc_has_vscale_cap);
+
 #define DPU_UNIT_INIT(dpu, base, unit, name, id, pec_ofs, ofs)		\
 {									\
 	int ret;							\
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 1b0660e..2b4730b 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -600,6 +600,9 @@ void layerblend_pixengcfg_clken(struct dpu_layerblend *lb,
 struct dpu_hscaler *fetchdecode_get_hscaler(struct dpu_fetchdecode *fd);
 struct dpu_vscaler *fetchdecode_get_vscaler(struct dpu_fetchdecode *fd);
 
+bool dpu_vproc_has_hscale_cap(u32 cap_mask);
+bool dpu_vproc_has_vscale_cap(u32 cap_mask);
+
 /*
  * to avoid on-the-fly/hot plane resource migration
  * between two display interfaces
-- 
1.7.9.5

